Update macOS dev docs (#16625)
opf/openproject: docs/development/development-environment/macos/README.md

default: &default
  adapter: postgresql
  encoding: unicode
  # Socket encryption must be disabled on macOS. There is an old bug in which forked processes cause problems on a Mac.
  # TL;DR: set this flag, otherwise Ruby will crash whenever a route is accessed.
  # Visit https://github.com/ged/ruby-pg/issues/311 to enter the rabbit hole.

>   gssencmode: disable

Reviewer: Is this preferred over setting it on your shell profile with `export PGGSSENCMODE="disable"`? If I recall correctly, this is only a problem for Apple Silicon machines.

You might also run into other Apple Silicon issues that can be solved with adding `export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=yes` to your shell profile

Reviewer 2: I think the `gssencmode` issue predates Apple Silicon. [This issue](https://github.com/ged/ruby-pg/issues/311) was reported in 2019, before the M1 was released.

Good hint about `OBJC_DISABLE_INITIALIZE_FORK_SAFETY`, do you remember which issues you encountered before you exported this?
*rumbling noises* ... I just peeked into a veeeery deep rabbit hole looking into this. It does connect to forking processes and Spring. It was allegedly fixed in Ruby 2.x same as the gssencmode trouble. So this keeps coming back 😉 

Some little postcards from the rabbit hole:
* https://github.com/rails/rails/issues/38560
* https://bugs.ruby-lang.org/issues/14009#note-4

All in all, I think it's definitely worth it to add `OBJC_DISABLE_INITIALIZE_FORK_SAFETY` to this guide. Additionally, mentioning the possibility of configuring these flags via env variable is a nice courtesy.

I will add a commit ✍🏻

Reviewer: > Some little postcards from the rabbit hole:
> https://github.com/rails/rails/issues/38560

Is where I first stumbled upon a couple months ago

the `.env.example` to `.env` and add the environment variables you want to configure. The variables will be
automatically loaded to the application's environment.

### Install other requirements

Some operations in OpenProject require additional software. These include PDF exports and interaction with an
LDAP server.

> In order to have OpenProject behave correctly, you need to install `imagemagick` and `java`.

Reviewer 3: I would write

> In order to develop OpenProject in these areas, you need to install `imagemagick` and `java`.

Just a note, but OP behaves still correctly without the optional requirements, but of course without the features. 
As you explained in the chat, Java is only required if users want to develop something with LDAP, not running it for simple development.

Please ignore and resolve if it's too nitpicky.

the `.env.example` to `.env` and add the environment variables you want to configure. The variables will be
automatically loaded to the application's environment.

> [!TIP]
> Instead of using the `gssencmode` flag in `config/database.yml`, you can add `export PGGSSENCMODE="disable"` to your
> Shell profile (`~/.zprofile` by default). This will prevent Ruby crashes for *all* of your projects.
>
> However, some users report Ruby crashes despite having setting this flag to disable.

> > If this happens on your machine, try adding `export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=yes` to your Shell profile.

Reviewer: ```suggestion
If this is the case for you as well, try adding `export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=yes` to your Shell profile.
```

?
